pkg/cri/resource-manager/cache/cache.go

```diff
@@ -694,6 +696,35 @@ func (cch *cache) LookupContainer(id string) (Container, bool) {
 	return c, ok
 }
 
+// LookupContainerByCgroup looks up the container for the given cgroup path.
+func (cch *cache) LookupContainerByCgroup(path string) (Container, bool) {
+	cch.Debug("resolving %s to a container...", path)
+
+	for id, c := range cch.Containers {
+		if id != c.CacheID {
+			continue
+		}
+
+		parent := ""
+		if pod, ok := c.GetPod(); ok {
+			parent = pod.GetCgroupParentDir()
+		}
+		if parent == "" {
+			continue
+		}
+
+		if !strings.HasPrefix(path, parent) {
+			continue
+		}
+
+		if strings.Index(path, c.GetID()) != -1 {
+			return c, true
+		}
+	}
+
+	return nil, false
```

Comment:
Would `nil, true` be valid in some case? Could we just return `Container`?

Reply:
> Would `nil, true` be valid in some case? Could we just return `Container`?

I think this is the classical go nil interface problem. If we returned (nil, true) here, how could you tell a valid Container apart from a nil Container interface without using reflect ?

Reply:
> I think this is the classical go nil interface problem. If we returned (nil, true) here, how could you tell a valid Container apart from a nil Container interface without using reflect ?

Yeah, that's true. What I was thinking was that is it a "legitimate" for these functions to return a `nil` Container interface in any condition?